samples/client.c

```diff
@@ -535,15 +536,23 @@ set_migration_state(int sock, uint32_t state)
             .iov_len = sizeof(change_state)
         }
     };
+    void* response = malloc(sizeof(req) + sizeof(change_state));
     
+    pthread_mutex_lock(&mutex);
     int ret = tran_sock_msg_iovec(sock, msg_id--, VFIO_USER_DEVICE_FEATURE,
                                   send_iovecs, 3, NULL, 0, NULL,
-                                  &req, sizeof(req), NULL, 0);
+                                  response, sizeof(req) + sizeof(change_state),
+                                  NULL, 0);
+    pthread_mutex_unlock(&mutex);
 
     if (ret < 0) {
         return -1;
     }
 
+    assert(memcmp(&req, response, sizeof(req)) == 0);
```

Comment:
Are you checking that the server succesfully changed migration state? If so then it should not be done with `assert` and we should be printing relevant values in case of error as this helps debugging.